### PR TITLE
Add Model Tensor Planning (MTP) controller

### PR DESCRIPTION
**Pull request by anindex**
Add the MTP sampling-based MPC controller (Le et al. 2025, arxiv 2505.01059).

MTP mixes two sampling strategies:
- **Tensor branch**: draws trajectories by uniformly sampling M waypoints within actuator bounds, then smoothing via Akima, B-spline, or linear interpolation (using the spline primitives from #74).
- **Local branch**: CEM-style Gaussian perturbations around the current mean, with softmax-weighted elite updates and momentum smoothing.

The `beta` parameter controls the tensor/local split. One slot is always reserved for the previous-best trajectory as a warm start.

### Files

- `hydrax/algs/mtp.py` - controller implementation
- `tests/test_mtp.py` - open-loop pendulum swingup, shape checks across all interpolation modes, validation guards
- `hydrax/algs/__init__.py` - export `MTP`
- `README.md` - add MTP to methods table

Depends on #74 (merged).

### CODE REVIEW

File: hydrax/algs/mtp.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+        """Refit the local Gaussian with weighted elite statistics."""
+        costs = jnp.sum(rollouts.costs, axis=1)  # (num_samples,)
+
+        # Top-K elite selection.
+        _, elite_idx = jax.lax.top_k(-costs, self.num_elites)
+        elite_knots = rollouts.knots[elite_idx]
+        elite_costs = costs[elite_idx]
+
+        # Softmax weighting with baseline subtraction for numerical stability.
+        baseline = jnp.min(elite_costs)
```
Comment (vincekurtz) [inline hydrax/algs/mtp.py]: jax.nn.softmax implements the baseline trick internally: there's no need to recompute it manually here.

File: hydrax/algs/mtp.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        var = jnp.sum(
+            weights[:, None, None] * (elite_knots - mean) ** 2, axis=0
+        )
+        bessel = 1.0 / jnp.maximum(1.0 - jnp.sum(weights**2), 1e-6)
+        cov = jnp.sqrt(var * bessel)
+        cov = jnp.clip(cov, self.sigma_min, self.sigma_max)
+
+        # Momentum smoothing: convex blend of old and new variance.
+        mean = mean + self.alpha * (params.mean - mean)
+        new_var = cov**2
```
Comment (vincekurtz) [inline hydrax/algs/mtp.py]: I would make the covariance be the standard deviation squared. That way you don't need to worry about squaring and square-rooting repeatedly

File: hydrax/algs/mtp.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        old_var = params.cov**2
+        cov = jnp.sqrt(self.alpha * old_var + (1.0 - self.alpha) * new_var)
```
Comment (vincekurtz) [inline hydrax/algs/mtp.py]: Why not clip the covariance here, after the smoothing?

**Review by vincekurtz — CHANGES_REQUESTED** [review]
A few small comments

File: hydrax/algs/mtp.py (hunk 1 above)
Comment (anindex) [reply hydrax/algs/mtp.py]: nice catch! done.

**Comment by anindex** [comment]
Thanks for the nice catches! These are important tweaks.

**Review by vincekurtz — APPROVED** [review]
LGTM
